### PR TITLE
feat : Added Products to products page

### PR DESCRIPTION
**Pull request by arjun-computer-geek**
Refactor the code : 
- made separate reducers and actions

files to check:
/actions/productAction.js
/reducers/productReducers.js
/context/productContext.js
/pages/categories.js

### CODE REVIEW

**Review by Shubhamkr2610 — APPROVED** [review]
Your logic is good ,  code is clean ,readable  and understandable. you are good to go

**Review by saurabhchirde — APPROVED** [review]
Clean and easy to read code.  Couldn't find any issue. 
Good to merge.

File: frontend/src/pages/categories/Categories.js [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
+  useEffect(() => {
+    getProducts(dispatch);
+  }, []);
+  const { loading, products, error } = state;
+  return (
+    <>
+      <main className="container">
         <Sidebar />
         <div className="main-content">
           {products.map((product, index) => (
```
Comment (Developers-codz) [inline frontend/src/pages/categories/Categories.js:19]: you can de-structure the product object. This way repeatition of product can be avoided

File: frontend/src/reducers/productReducers.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  ALL_PRODUCTS_SUCCESS,
+  ALL_PRODUCTS_FAIL,
+} from "constants/productContants";
+
+export const productReducer = (state = { products: [] }, action) => {
+  switch (action.type) {
+    case ALL_PRODUCTS_REQUEST:
+      return {
+        loading: true,
+        products: [],
```
Comment (Developers-codz) [inline frontend/src/reducers/productReducers.js:12]: don't understood why u have de-structured only products and not loading and using loading without `state.loading` if it is working fine then u don't need to de-structure it

**Review by Developers-codz — APPROVED** [review]
Else code looks good, but do provide screenshots or deploy your site. It will be easy to review for us

File: frontend/src/reducers/productReducers.js (hunk 2 above)
Comment (arjun-computer-geek) [reply frontend/src/reducers/productReducers.js:12]: Soryy but I am not destructing the product.

Comment (arjun-computer-geek) [reply frontend/src/reducers/productReducers.js:12]: It is just returning the empty product and loading to true. Loading is for set the loader.

**Pull request merged**